tests/grpc/grpc_test.go

```diff
@@ -82,6 +89,28 @@ func startGrpcListener(t *testing.T, r *task.TaskRegistry) context.CancelFunc {
 	return cancel
 }
 
+func Test_Grpc_WaitForInstanceStart_Timeout(t *testing.T) {
+	r := task.NewTaskRegistry()
+	r.AddOrchestratorN("WaitForInstanceStartThrowsException", func(ctx *task.OrchestrationContext) (any, error) {
+		// sleep 5 seconds
+		time.Sleep(5 * time.Second)
+		return 20, nil
+	})
+
+	cancelListener := startGrpcListener(t, r)
+	defer cancelListener()
+
+	id, err := grpcClient.ScheduleNewOrchestration(ctx, "WaitForInstanceStartThrowsException", api.WithInput("世界"))
+	require.NoError(t, err)
+	timeoutCtx, cancelTimeout := context.WithTimeout(ctx, time.Second)
+	defer cancelTimeout()
+	_, err = grpcClient.WaitForOrchestrationStart(timeoutCtx, id, api.WithFetchPayloads(true))
+	if assert.Error(t, err) {
+		assert.Contains(t, err.Error(), "context deadline exceeded")
+	}
+	time.Sleep(1 * time.Second)
```

Comment:
Can we add an another test(or modify this one) as well where we close the stream, wait for sometime and then re-connect. The orchestration which was scheduled previously and couldn't complete should complete now..

Reply:
I think it's a good test to add. Just a caveat that the WI would have to be restarted.

Reply:
Do you mean restarting the work item? Since we are returning an error from executor in case stream gets closed, the current work item would be abandoned and should be retried once we have the active stream again. right?

Reply:
I probably didn't do a good job writing what I meant to say :) Just that when the app comes back online, it needs to fetch the work items again. It can't assume that it can resume processing whatever WI it was processing when it died

Reply:
added test.